File: vfoot-frontend/src/utils/currency.ts
// La valuta della lega, in un posto solo.
//
// Prima non esisteva un posto solo, ed è per questo che dentro la STESSA scheda
// rosa il prezzo di un giocatore era «€28» e il totale del reparto due righe
// sopra «48 crediti»: due parole per la stessa cosa, scelte da due persone in
// due momenti. Un euro poi è una moneta vera, e nel listone ce n'è davvero uno —
// il valore di mercato Transfermarkt — quindi lo stesso segno diceva due cose
// diverse a due centimetri di distanza.
//
// IL SIMBOLO E LA PAROLA FANNO DUE MESTIERI. Il simbolo sta attaccato a un
// numero, dove lo spazio è quello di una colonna e nessuno legge, guarda:
// `ṿƒ28`. La parola serve dove si sta parlando — «hai 137 vfooties», «offerta in
// vfooties» — e lì un glifo di due caratteri sarebbe un rebus. Non sono
// intercambiabili, e le due funzioni qui sotto esistono per non doverci pensare
// ogni volta.
export const CURRENCY_SYMBOL = 'ṿƒ';
export const CURRENCY_NAME = 'vfooty';
export const CURRENCY_NAME_PLURAL = 'vfooties';

/** Un prezzo col suo simbolo dopo: `28 ṿƒ`. Per colonne e tabelle.
 *
 *  DOPO e staccato, come si scrive un prezzo parlando italiano (30 €, non € 30):
 *  in una colonna allineata a destra il numero resta il primo a essere letto e le
 *  cifre restano incolonnate, mentre il simbolo davanti spingeva i numeri a
 *  destra di larghezze diverse a seconda che fossero due o tre cifre. */
export function price(amount: number): string {
  return `${Math.round(amount)} ${CURRENCY_SYMBOL}`;
}

/** Una cifra detta a parole: `28 vfooties`. Per le frasi. */
export function amount(value: number): string {
  const n = Math.round(value);
  return `${n} ${n === 1 ? CURRENCY_NAME : CURRENCY_NAME_PLURAL}`;
}

File: vfoot-backend/src/vfoot/services/currency.py
"""La valuta della lega, dal lato del server.

Il gemello di ``vfoot-frontend/src/utils/currency.ts``, e serve perché una parte
dei testi che parlano di soldi NON li scrive il browser: la riga di news di un
acquisto e i messaggi che rifiutano un'offerta nascono qui, e prima dicevano
"crediti" mentre la pagina accanto diceva "€". Due copie di una costante sono
poco eleganti, ma l'alternativa — mandare la parola al browser a ogni richiesta —
sarebbe un campo in più in ogni risposta per un valore che non cambia mai.

Il simbolo sta attaccato ai numeri, la parola sta nelle frasi: sono due mestieri
diversi, e le due funzioni esistono per non doverci pensare ogni volta.
"""
from __future__ import annotations

SYMBOL = "ṿƒ"
NAME = "vfooty"
NAME_PLURAL = "vfooties"


def price(value) -> str:
    """`28 ṿƒ` — il simbolo DOPO e staccato, come si scrive un prezzo in italiano."""
    return f"{int(round(float(value)))} {SYMBOL}"


def amount(value) -> str:
    """`28 vfooties` — per stare dentro una frase."""
    n = int(round(float(value)))
    return f"{n} {NAME if n == 1 else NAME_PLURAL}"
